src/ui/states/state-present-offer.svelte

```diff
@@ -1,43 +1,58 @@
 <script lang="ts">
     import ModalSection from "../modal-section.svelte";
     import { getRenewsLabel, getTrialsLabel } from "../../helpers/price-labels";
-    import type { Product, PurchaseOption, SubscriptionOption } from "../../entities/offerings";
+    import {
+        type NonSubscriptionOption,
+        type Product,
+        ProductType,
+        type PurchaseOption,
+        type SubscriptionOption,
+    } from "../../entities/offerings";
 
     export let productDetails: Product;
     export let purchaseOption: PurchaseOption;
 
+    const isSubscription = productDetails.productType === ProductType.Subscription;
     const subscriptionOption: SubscriptionOption | null | undefined =
       purchaseOption as SubscriptionOption;
-    const trial = subscriptionOption?.trial;
-    const basePrice = subscriptionOption?.base?.price;
+    const nonSubscriptionOption: NonSubscriptionOption | null | undefined =
+      purchaseOption as NonSubscriptionOption;
```

Comment:
I would really love to have something like sealed/final classes in typescript but I couldn't find anything :(. Happy to receive feedback if there are better ways to do this! cc @guido732

Reply:
hmmmm that'd be useful but there's not an easy way to do so. I did some research and there are workarounds but nothing integrated that could be used like this.

What's the end goal you're trying to reach in this specific case? why would you want to lock the class?

Reply:
Right, so in Kotlin/Swift you can use sealed classes/enum classes to do something like this:
```
sealed class PurchaseOption {
  data class SubscriptionOption(...): PurchaseOption
  data class NonSubscriptionOption(...): PurchaseOption
}
```
Then, later in your code:
```
when (purchaseOption) {
  is SubscriptionOption -> { /* Here the compiler ensures it's a SubscriptionOption */ }
  is NonSubscriptionOption -> { /* Here the compiler ensures it's a NonSubscriptionOption */ }
}
```

It's very useful for polymorphic classes since it ensures you handle the different types enforcedly by the compiler. 

But I believe this doesn't exist in typescript since most things are simple interfaces and we only check for conformance to those interfaces... With the current code I have, if we added a new subtype, the typescript compiler wouldn't fail and we would need to remember to update it. Probably not a huge deal and we can live with it, but just wondering if you had better ideas as for how to organize the code.